client/opentracing/src/test/java/org/hawkular/apm/client/opentracing/TraceContextTest.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright 2015-2016 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.hawkular.apm.client.opentracing;
+
+import static org.junit.Assert.assertEquals;
+
+import org.hawkular.apm.api.utils.PropertyUtil;
+import org.junit.Before;
+import org.junit.Test;
+
+/**
+ * @author Juraci Paixão Kröhling
+ */
+public class TraceContextTest {
+
+    @Before
+    public void cleanupPossibleConflictingEnvVars() {
+        System.getProperties().remove(PropertyUtil.HAWKULAR_APM_SERVICE_NAME);
+        System.getProperties().remove(PropertyUtil.OPENSHIFT_BUILD_NAME);
+        System.getProperties().remove(PropertyUtil.OPENSHIFT_BUILD_NAMESPACE);
+    }
+
+    @Test
+    public void convertToServiceName_usualCase() {
+        assertEquals("hawkular-apm", TraceContext.getServiceFromBuildName("hawkular-apm-1"));
+    }
+
+    @Test
+    public void convertToServiceName_reallyLargeBuildNumber() {
+        assertEquals("hawkular-apm", TraceContext.getServiceFromBuildName("hawkular-apm-111111111111111111111111111111111111111111"));
+    }
+
+    @Test
+    public void convertToServiceName_withoutNumber() {
+        // this looks wrong, but we have to decide what to do here: in *all* currently known situations, the build name is a composition
+        // of {service_name}-{build_number}, so, we won't try to be too smart here... This test is here to document this conscious decision
+        assertEquals("hawkular", TraceContext.getServiceFromBuildName("hawkular-apm"));
+    }
+
+    @Test
+    public void convertToServiceName_empty() {
+        assertEquals("", TraceContext.getServiceFromBuildName(""));
+    }
+
+    @Test
+    public void convertToServiceName_nil() {
+        assertEquals(null, TraceContext.getServiceFromBuildName(null));
+    }
+
+
+    @Test
+    public void retrieveServiceName_noExplicitPropertyOutsideOpenShift() {
+        // this is probably the most common case: there's no service name and we can't guess one, so, we just return null.
+        System.getProperties().remove(PropertyUtil.HAWKULAR_APM_SERVICE_NAME);
```

Comment:
Its not really a problem, just redundant as called in `cleanupPossibleConflictingEnvVars` before each test.